pkg/migrations/backfill/backfill.go

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package backfill
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+
+	"github.com/xataio/pgroll/pkg/schema"
+)
+
+// BackFill updates all rows in the given table, in batches, using the
+// following algorithm:
+// 1. Get the primary key column for the table.
+// 2. Get the first batch of rows from the table, ordered by the primary key.
+// 3. Update each row in the batch, setting the value of the primary key column to itself.
+// 4. Repeat steps 2 and 3 until no more rows are returned.
+func BackFill(ctx context.Context, conn *sql.DB, table *schema.Table) error {
+	// Get the primary key column for the table
+	pks := table.GetPrimaryKey()
+	if len(pks) != 1 {
+		return errors.New("table must have a single primary key column")
```

Comment:
is this a hard requirement? multi column primary keys are not that strainge

Reply:
I think the same approach could work with multi-column PKs, but creating multi-column PKs is not currently supported by `pgroll`. 

I'd rather leave it with this limitation in place and then test the multi-column PK behaviour when such tables are supported by `pgroll`. At the moment it's hard to write a test for as we can't set up such a table with `pgroll` migrations.

Reply:
that sounds good for now, although pgroll can work against existing schemas. I believe at the very least we should detect this situation in validate so we don't attempt to start a migration that will result on this error

Reply:
I've added extra validation checks for this case.